File: app/assets/javascripts/models/guest.js
App.Guest = DS.Model.extend({

  name:     DS.attr('string'),
  response: DS.attr('string'),
  party:    DS.belongsTo('party', { async: true }),
  meal:     DS.belongsTo('meal', { async: true }),

  isAttending: function () {
    return this.get('response') === 'attending';
  }.property('response'),

  isNotAttending: function () {
    return this.get('response') === 'not_attending';
  }.property('response'),

  clearMeal: function () {
    if (this.get('response') !== 'attending') {
      this.set('meal', null);
    }
  }.observes('response')

});

File: app/models/guest.rb
class Guest < ActiveRecord::Base

  belongs_to :party
  belongs_to :meal

  validates :response, :inclusion => { :in   => ["attending", "not_attending"],               :message => "must be filled out" }, :on => :update
  validates :email,    :format    => { :with => /\A[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}\z/, :message => "isn't valid" },        :unless => "email.blank?"

  validates :meal,     :presence  => { :message => "can't be blank if you're attending" },    :if     => "attending?"
  validates :meal,     :absence   => { :message => "must be blank if you're not attending" }, :unless => "attending?"

  enum :response => [ :unconfirmed, :attending, :not_attending ]

end
